File: Strings/__3000__/3504/3504.py
sentense = input()
ptr1 = 0
ptr2 = len(sentense) - 1
isPalindrome = True

while ptr1 != ptr2:
    if sentense[ptr1] != sentense[ptr2]:
        isPalindrome = False
        break
    ptr1 += 1
    ptr2 -= 1

if isPalindrome:
    print(f"A frase [{sentense}] eh palindrome")
else:
    print(f"A frase [{sentense}] nao eh palindrome")
    

File: Strings/__3000__/3504/3504.rs
use std::io;

fn main() {
    let mut input = String::new();
    io::stdin().read_line(&mut input).unwrap();

    // Remove '\n' e '\r'
    let input = input.trim();

    let bytes = input.as_bytes();

    let mut i: usize = 0;
    let mut f: usize = bytes.len() - 1;
    let mut is_palindrome = true;

    while i < f {
        if bytes[i] != bytes[f] {
            is_palindrome = false;
            break;
        }
        i += 1;
        f -= 1;
    }

    if is_palindrome {
        println!("A frase [{}] eh palindrome", input);
    } else {
        println!("A frase [{}] nao eh palindrome", input);
    }
}